src/serializer/ResidualFunctions.js

```diff
@@ -288,19 +291,46 @@ export class ResidualFunctions {
       // rewritten functions shouldn't have references fixed up because the body,
       // consists of serialized code. For simplicity we emit their instances in a naive way
       let functionBody = t.blockStatement(rewrittenBody);
-
+      let funcParams = params.slice();
+      let funcOrClassNode;
+      let { classProperties } = instance;
+
+      if (classProperties !== undefined) {
+        let { methodType, classMethodKeyNode, classSuperNode, classMethodComputed } = classProperties;
+
+        let isConstructor = methodType === "constructor";
+        let homeObject = functionValue.$HomeObject;
+        invariant(homeObject instanceof ObjectValue);
+        invariant(classMethodKeyNode && (t.isExpression(classMethodKeyNode) || t.isIdentifier(classMethodKeyNode)));
+        // we use the $HomeObject as the key to get the class expression ast node
+        funcOrClassNode = this._getOrCreateClassNode(homeObject);
+        let classMethod = t.classMethod(methodType, classMethodKeyNode, funcParams, functionBody, classMethodComputed);
+        // add the class method to the class expression node body
+        if (isConstructor) {
+          funcOrClassNode.body.body.unshift(classMethod);
+        } else {
+          funcOrClassNode.body.body.push(classMethod);
+        }
```

Comment:
Just because this method was originally defined on this class, doesn't mean it still is the same one.

E.g.

```js
class Foo {
  bar() {
  }
  baz() {
  }
}
Foo.prototype.bar = Foo.prototype.bar.bind({});
var baz = Foo.prototype.baz;
delete Foo.prototype.baz;
result = { Foo, baz };
```

We should test that the property on the class is still the correct one.

It's probably fine not supporting this yet, but we should at least fatal and not generate subtly wrong code.

Reply:
Static class methods also have home objects and will end up here but they will be serialized as an instance method instead which is wrong.

Reply:
Static methods are supported fine – they were tested. They are actually properties of the constructor function.

Reply:
If so, can you add a unit test for it? With and without a super call in the static method.

Reply:
@sebmarkbage Sure. Although I'm not sure what is special about the `super` call. Prepack doesn't touch it as we don't ever evaluate it.

Reply:
Super is special because that can't be serialized as:

```js
class Foo extends Bar {}
Foo.bar = function() { super(); };
```

Reply:
This is fixed now.

Reply:
Is there a test case for this?